plugins/xpressive/exprfront.cpp

```diff
@@ -0,0 +1,610 @@
+0x28ac8b2b, 0x10f779b2, 0x6e79e659 ,0x32e44c52,
+0xf790aa55, 0x98b05083, 0xb5d44f1c ,0xe553da04,
+0xa884c6d2, 0x43274953, 0xbcb57404 ,0x43f7d32a,
+0xf1890f8b, 0x019f4dce, 0x5c4ede33 ,0x2dec1a7e,
+0x0f3eab09, 0x2197c93c, 0xae933f42 ,0x80d4b111,
+0x6e5bd30a, 0x17139c70, 0xe15f7eb0 ,0x1798f893,
+0xe1c6be1c, 0xe21edf9b, 0x4702e081 ,0x8a2cb85a,
+0xbf3c1f15, 0x147f4685, 0x9221d731 ,0x3c7580f3,
+0xc1c08498, 0x8e198b35, 0xf821c15a ,0x4d3cd2d4,
+0xad89a3b7, 0xd48f915f, 0xcaf893f0 ,0xa64a4b8e,
+0x20715f54, 0x1ba4de0a, 0x17ac6e91 ,0xd82ea8c0,
+0x638a0ba5, 0xe7a76c0f, 0x486c5476 ,0x334bbd0a,
+0xffe29c55, 0x7247efaf, 0x15f98e83 ,0x7a4a79ac,
+0x350cd175, 0xc7107908, 0xa85c67f7 ,0x9c5002c4,
+0x3cf27d2c, 0x314d8450, 0x05552886 ,0x87a73642,
+0x827832e4, 0x9412cc67, 0x261979e6 ,0xb31da27f,
+0x3e6bbafb, 0x663f1968, 0xd84274e2, 0xdd91d982,
+0xd25c4805, 0x9567f860, 0xab99675c, 0x2254733b,
+0x18799dd7, 0xee328916, 0xb9419a1b, 0x01b7a66f,
+0xbcdc05e1, 0x788de4ae, 0x366e77cf, 0x81a1ebd2,
+0x97be859a, 0x17d4b533, 0x22dab3a9, 0xc99871ea,
+0xc7502c91, 0x4474b65f, 0x655d059d, 0x0ddc1348,
+0x8325909b, 0x4873c155, 0x9fa30438, 0x7250b7a8,
+0x90db2715, 0xf65e1cef, 0x41b74cf4, 0x38fba01c,
+0xe9eefb40, 0x9e5524ea, 0x1d3fc077, 0x04ec39db,
+0x1c0d501c, 0xb93f26d9, 0xf9f910b9, 0x806fce99,
+0x5691ffdf, 0x1e63b27a, 0xf2035d42, 0xd3218a0b,
+0x12eae6db, 0xeba372a9, 0x6f975260, 0xc514ae91,
+0xebddb8ad, 0xc53207c0, 0xdbda57dc, 0x8fb38ef4,
+0xfaa4f1bc, 0x40caf49f, 0xcb394b41, 0x424fc698,
+0xb79a9ece, 0x331202d6, 0xc604ed4d, 0x5e85819f,
+0x67222eda, 0xd976ba71, 0x7d083ec6, 0x040c819e,
+0xc762c934, 0xa6684333, 0x2eaccc54, 0x69dc04b9,
+0x0499cf36, 0x6351f438, 0x6db2dc34, 0x787ae036,
+0x11b5c6ac, 0x552b7227, 0x32a4c993, 0xf7f4c49d,
+0x7ac9e2d9, 0xf3d32020, 0x4ff01f89, 0x6f0e60bb,
+0x3c6ed445, 0x7ca01986, 0x96901ecf, 0xe10df188,
+0x62a6da6d, 0x8deee09f, 0x5347cb66, 0x5249f452,
+0x22704d4d, 0x6221555f, 0x6aa0ea90, 0xe1f7bae3,
+0xd106626f, 0x6365a9db, 0x1989bb81, 0xfc2daa73,
+0x303c60b3, 0xcd867baa, 0x7c5787c2, 0x60082b30,
+0xa68d3a81, 0x15a10f5d, 0x81b21c8a, 0x4bfb82e2,
+0xff01c176, 0xff3c8b65, 0x8cc6bd29, 0xc678d6ff,
+0x99b86508, 0x3c47e314, 0x766ecc05, 0xba186cb0,
+0x42f57199, 0x5ef524f4, 0xb8419750, 0x6ae2a9d0,
+0x291eaa18, 0x4e64b189, 0x506eb1d3, 0x78361d46,
+0x6a2fcb7e, 0xbc0a46de, 0xb557badf, 0xad3de958,
+0xa2901279, 0x491decbf, 0x257383df, 0x94dd19d1,
+0xd0cfbbe2, 0x9063d36d, 0x81e44c3b, 0x973e9cc9,
+0xfbe34690, 0x4eee3034, 0x1c413676, 0xf6735b8f,
+0xf2991aca, 0x0ec85159, 0x6ce00ade, 0xad49de57,
+0x025edf30, 0x42722b67, 0x30cfa6b2, 0x32df8676,
+0x387d4500, 0x97fa67fd, 0x027c994a, 0x77c71d0c,
+0x478eb75a, 0x898370a6, 0x73e7cca3, 0x34ace0ad,
+0xc8ecb388, 0x5375c3aa, 0x9c194d87, 0x1b65246d,
+0xca000bcf, 0x8a0fb13d, 0x81b957b0, 0xac627bfb,
+0xc0fe47e5, 0xf3db0ad8, 0x1c605c7d, 0x5f579884,
+0x63e079b5, 0x3d96f7cf, 0x3edd46e9, 0xc347c61e,
+0x7b2b2a0e, 0x63dfcf51, 0x596781dd, 0x80304c4d,
+0xa66f8b47
+};
+
+inline unsigned int rotate(unsigned int x, const int b)
```

Comment:
You should call this `rotateLeft`. Also, you should check if `b == 0` because you could cause undefined behavior.